lib/Model.php

```diff
@@ -1623,94 +1590,200 @@ public function __call($method, $args)
-     * Get a count of qualifying records.
+     * @see Relation::joins()
      *
-     * ```
-     * YourModel::count(['conditions' => 'amount > 3.14159265']);
-     * ```
+     * @param string|array<string> $joins
      *
-     * @see find
+     * @return Relation<static>
+     */
+    public static function joins(string|array $joins): Relation
+    {
+        return static::Relation()->joins($joins);
+    }
+
+    /**
+     * @see Relation::order()
      *
-     * @return int Number of records that matched the query
+     * @return Relation<static>
      */
-    public static function count(/* ... */): int
+    public static function order(string $order): Relation
     {
-        $args = func_get_args();
-        $options = static::extract_and_validate_options($args);
-        $options['select'] = 'COUNT(*)';
+        return static::Relation()->order($order);
+    }
 
-        if (!empty($args) && !is_null($args[0]) && !empty($args[0])) {
-            if (is_hash($args[0])) {
-                $options['conditions'] = $args[0];
-            } else {
-                $options['conditions'] = static::pk_conditions(...$args);
-            }
-        }
+    /**
+     * @see Relation::group()
+     *
+     * @return Relation<static>
+     */
+    public static function group(string $columns): Relation
+    {
+        return static::Relation()->group($columns);
+    }
 
-        $table = static::table();
-        $sql = $table->options_to_sql($options);
-        $values = $sql->get_where_values();
+    /**
+     * @see Relation::limit()
+     *
+     * @return Relation<static>
+     */
+    public static function limit(int $limit): Relation
+    {
+        return static::Relation()->limit($limit);
+    }
+
+    /**
+     * @see Relation::having()
+     *
+     * @return Relation<static>
+     */
+    public static function offset(int $offset): Relation
+    {
+        return static::Relation()->offset($offset);
+    }
 
-        $res = static::connection()->query_and_fetch_one($sql->to_s(), $values);
+    /**
+     * @see Relation::having()
+     *
+     * @return Relation<static>
+     */
+    public static function having(string $having): Relation
+    {
+        return static::Relation()->having($having);
+    }
 
-        return $res;
+    /**
+     * @see Relation::having()
+     *
+     * @return Relation<static>
+     */
+    public static function from(string $from): Relation
+    {
+        return static::Relation()->from($from);
     }
 
     /**
-     * Determine if a record exists.
+     * @param string|array<string|mixed> $include
+     *
+     * @return Relation<static>
+     */
+    public static function include(string|array $include): Relation
+    {
+        return static::Relation()->include($include);
+    }
+
+    /**
+     * @see Relation::where()
+     *
+     * @return Relation<static>
+     */
+    public static function where(): Relation
+    {
+        return static::Relation()->where(...func_get_args());
+    }
+
+    /**
```

Comment:
These comments are stale and should be deleted.

Reply:
Right you are f54dba6